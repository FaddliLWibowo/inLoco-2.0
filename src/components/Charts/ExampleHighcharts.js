import React from 'react';
import Highcharts from 'highcharts';

/**
 * Componente ExampleHighcharts que representa um gráfico da biblioteca Highcharts.js
 */

export default class ExampleHighcharts extends React.Component {
    /**
     * cria o gráfico após renderizar o componente
     */
    componentDidMount () {
        // Extende o highcharts com módulos
        console.log(this);
        if (this.props.modules) {
            this.props.modules.forEach(function (module) {
                module(Highcharts);
            });
        }
        // Cria o gráfico passando o container onde o gráfico será renderizado
        this.chart = new Highcharts[this.props.type || "Chart"](
            this.props.container,
            this.props.options
        );
        console.log(this.chart);
    }
    /**
     * destrói o gráfico antes da remoção do componente
     */
    componentWillUnmount() {
        this.chart.destroy();
    }
    /**
     * renderiza o elemento que conterá o gráfico
     * @return html de marcação do elemento
     */
    render () {
        return (
            <div id={this.props.container} className="chart"></div>
        );
    }
}
