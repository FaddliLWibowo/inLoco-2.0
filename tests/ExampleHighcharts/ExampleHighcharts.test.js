import React from 'react';
import ExampleHighcharts from '../../src/components/Charts/ExampleHighcharts.js';
import renderer from 'react-test-renderer';

it('componente renderiza corretamente', () => {
    const exampleHighcharts  = shallow(<ExampleHighcharts/>);
    expect(exampleHighcharts).toMatchSnapshot();
});
