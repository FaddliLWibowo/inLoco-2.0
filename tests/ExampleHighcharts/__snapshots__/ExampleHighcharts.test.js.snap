// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componente renderiza corretamente 1`] = `
<div>
  <div
    className="chart"
  >
     sdsds
  </div>
</div>
`;
